### DISCUSSION TITLE
BaseBatteryModel() and Experiments()

### DISCUSSION THREAD

**Discussion by veera-visw** [pybamm-team/PyBaMM#3776] [General]
Hi! 

To give a brief intro on what I am tryin to achieve: I want modify the boundary conditions to the particle equation to account for side reactions. I have an idea of how to make it work starting from BaseModel() (for example basic_dfn.py). But I have had problems coupling Experiment class with the BaseBattery() [ Variable "Battery Voltage [V]" not found]

I am just looking for recommendations on what you think the right way to proceed is?

- Go with basic_dfn.py which is easier to modify, or try to use dfn.py?
- Am I correct in understanding that Experiment class is meant for predefined models, rather than the BaseModels()?

Thanks

**Comment by TomTranter** [comment]
I don't think you need to modify any of the base models directly but if you did you should look at the base_lithium_ion_model which subclasses base_battery. The best approach would be to dig into the SEI models and see how they are implemented then modify one of those

**Reply by veera-visw** [reply]
Sorry, yes I did mean the basebatterymodels like basic_spm, basic_dfn and so forth.

Will look at the SEI source code. Thanks for the suggestion.

**Comment by rtimms** [comment]
The experiments set up voltage and power controls using submodels that add an equation to solve that determines the current (see [here](https://github.com/pybamm-team/PyBaMM/blob/a09d6e7a7f13366d94b940bcd3064d68fde44d53/pybamm/simulation.py#L231)). To get the base models (e.g. basic DFN) to work with experiments you’ll need to make sure you define all the necessary variables to be compatible with those submodels. 

If you’re new to PyBaMM it is often simpler to modify the basic model first to get everything working before trying to add your new equations as a sub model.

**Reply by veera-visw** [reply]
I have looked at source code of SEI, and also at Basic_**.py models and I came to a similar conclusion. I will start with basic_spm.py and go from there ( i think I have a working model now to start with). Eventually, I will have to look at how to include this into the experiment class. 

I am not interested in Li plating, degradation etc.. that are common variables in the experiment class, so I would have to modify it to exclude such variables I suppose.  Thanks for suggestion, if I am stuck again, I will get back to this thread.

**Comment by veera-visw** [comment]
@rtimms  a quick follow up: basic_dfn.py --> is this, by itself, a complete model or was the idea to keep it as a framework to develop further?

1. I dont see the line where charge conservation is enforced?
2. The electrolyte concentration boundary conditions are specified as homogenous neumann on "left" and "right" --> does this left and right mean for each zones ? 

Just curious, I am still reading through the code, but might be a straightforward answer for you. Thanks

**Reply by veera-visw** [reply]
Just to close the topic, for anyone else who might be intersted, the charge conservation part is here:
![image](https://github.com/pybamm-team/PyBaMM/assets/157810026/dcd9cc16-9748-4b1c-94d3-26d70d5ed357)

It just took me a while to understand the difference between model.rhs() and model.algebraic(). Closing task
